### PR TITLE
Changes ApiController#load_script to return a cached script.

### PR DESCRIPTION
**Pull request by ashercodeorg**

### CODE REVIEW

File: dashboard/app/controllers/api_controller.rb [hunk 1]
```diff
@@ -350,7 +350,7 @@ def load_section
   end
 
   def load_script
-    @script = Script.get_from_cache(params[:script_id]) if params[:script_id].present?
-    @script ||= @section.script || Script.twenty_hour_script
+    script_id = params[:script_id] || @section.script.id
```
Comment (ashercodeorg) [inline dashboard/app/controllers/api_controller.rb]: Is it possible for `params[:script_id]` to be the empty string? If so, I need to change this line to skip over it.

**Comment by wjordan** [comment]
so this is an optimization that eliminates an extra query in the case where no `script_id` is provided and the script is derived from the current section, and I reading this right?

**Comment by ashercodeorg** [comment]
Correct. Though its impact is greater than the one extra query, as further associations on the `@script` variable will now also always be cached.

File: dashboard/app/controllers/api_controller.rb (hunk 1 above)
Comment (wjordan) [reply dashboard/app/controllers/api_controller.rb]: I don't know, but it seems that `"should get progress for section with section script"` test should cover the case where the `script_id` parameter is not present in any case.

**Comment by ashercodeorg** [comment]
Ugh, CircleCI failure are legitimate (`section.script` may be `nil`, so `section.script.id` is an error). Fixing...

**Comment by ashercodeorg** [comment]
The CircleCI failures seem to all be known flakiness or known bad head.

**Pull request merged**